roadmap/implementers-guide/src/runtime/disputes.md

```diff
@@ -17,90 +19,351 @@ We account for these requirements by having the validity module handle two kinds
+The local disputes are necessary in order to create the first escalation that leads to block producers abandoning the chain and making remote disputes possible.
 
-Second, we know off-chain when the approval process completes based upon all gossiped assignment notices, not just the approving ones.  We need not-yet-approved assignment notices to appear on-chain if the chain should know about the validity of recently approved blocks.  Relay chain blocks become eligible for finality in GRANDPA only once all their included candidates pass approvals checks, meaning all assigned checkers either voted approve or else were declared "no show" and replaced by more assigned checkers.  A purely off-chain approvals scheme complicates GRANDPA with additional objections logic.  
+Local disputes are only allowed on parablocks that have been included on the **local** chain and are in the acceptance period.
 
-Integration with GRANDPA appears simplest if we witness approvals in chain:  Aside from inherents for assignment notices and approval votes, we provide an "Approved" inherent by which a relay chain block declares a past relay chain block approved.  In other words, it trigger the on-chain approval counting logic in a relay chain block `R1` to rerun the assignment and approval tracker logic for some ancestor `R0`, which then declares `R0` approved.  In this case, we could integrate with GRANDPA by gossiping messages that list the descendent `R1`, but then map this into the approved ancestor `R0` for GRANDPA itself.
+For each such parablock, it is guaranteed by the inclusion pipeline that the parablock is available and the relevant validation code is available.
 
-Approval votes could be recorded on-chain quickly because they represent a major commitments.  
+Disputes may occur against blocks that have happened in the session prior to the current one, from the perspective of the chain.
 
-Assignment notices should be recorded on-chain only when relevant.  Any sent too early are retained but ignore until relevant by our off-chain assignment system.  Assignments are ignored completely by the dispute system because any dispute immediately escalates into all validators checking, but disputes count existing approval votes of course.
+In this case, the prior validator set is responsible for handling the dispute and to do so with their keys from the last session.
+This means that validator duty actually extends 1 session beyond leaving the validator set.
 
+Wait for a 2/3 majority either way, by counting incoming votes. Or timeout.
```

Comment:
I'm unsure why this comes under local as opposed to all disputes, but..

We slash if we reach 2/3 majority either way.  We declare the block invalid but do not slash if we fail to reach any 2/3 majority.